packages/browser-agent-core/common/window/load.js

```diff
@@ -0,0 +1,23 @@
+import { eventListenerOpts } from '../event-listener/event-listener-opts'
+
+const win = window
+const doc = win.document
+const ADD_EVENT_LISTENER = 'addEventListener'
+const ATTACH_EVENT = 'attachEvent'
+
+function checkState (cb) {
+  if (doc.readyState === 'complete') cb()
+}
+
+export function onWindowLoad(cb) {
+  checkState(cb)
+  if (doc[ADD_EVENT_LISTENER]) win[ADD_EVENT_LISTENER]('load', cb, eventListenerOpts(false))
+  else win[ATTACH_EVENT]('onload', cb)
+}
+
+export function onDOMContentLoaded(cb) {
+  checkState(cb)
```

Comment:
I think the comment above applies here too, `doc.readyState === 'complete'` could be any point after DOMContentLoaded. If we get to create a register before that event, we should call it, but if the event has already occurred we shouldn't call the callback immediately because it relies on `now()` and assumes we are at the moment of DOMContentLoaded.

Reply:
https://github.com/newrelic/newrelic-browser-agent/pull/255#discussion_r934991359